code/modules/mob/living/simple_animal/guardian/types/assassin.dm

```diff
@@ -1,99 +1,7 @@
-	..()
-	stealthcooldown = 0
-
-/mob/living/simple_animal/hostile/guardian/assassin/Life()
-	. = ..()
-	updatestealthalert()
-	if(loc == summoner && toggle)
-		ToggleMode(0)
-
-/mob/living/simple_animal/hostile/guardian/assassin/Stat()
-	..()
-	if(statpanel("Status"))
-		if(stealthcooldown >= world.time)
-			stat(null, "Stealth Cooldown Remaining: [max(round((stealthcooldown - world.time)*0.1, 0.1), 0)] seconds")
-
-/mob/living/simple_animal/hostile/guardian/assassin/AttackingTarget()
-	if(..())
-		if(toggle && (isliving(target) || istype(target, /obj/structure/window) || istype(target, /obj/structure/grille)))
-			ToggleMode(1)
-
-/mob/living/simple_animal/hostile/guardian/assassin/adjustHealth(amount, updating_health = TRUE, forced = FALSE)
-	. = ..()
-	if(. > 0 && toggle)
-		ToggleMode(1)
-
-/mob/living/simple_animal/hostile/guardian/assassin/Recall()
-	if(..() && toggle)
-		ToggleMode(0)
-
-/mob/living/simple_animal/hostile/guardian/assassin/ToggleMode(forced = 0)
-	if(toggle)
-		melee_damage_lower = initial(melee_damage_lower)
-		melee_damage_upper = initial(melee_damage_upper)
-		armour_penetration = initial(armour_penetration)
-		obj_damage = initial(obj_damage)
-		environment_smash = initial(environment_smash)
-		alpha = initial(alpha)
-		if(!forced)
-			src << "<span class='danger'><B>You exit stealth.</span></B>"
-		else
-			visible_message("<span class='danger'>\The [src] suddenly appears!</span>")
-			stealthcooldown = world.time + initial(stealthcooldown) //we were forced out of stealth and go on cooldown
-			cooldown = world.time + 40 //can't recall for 4 seconds
-		updatestealthalert()
-		toggle = FALSE
-	else if(stealthcooldown <= world.time)
-		if(src.loc == summoner)
-			src << "<span class='danger'><B>You have to be manifested to enter stealth!</span></B>"
-			return
-		melee_damage_lower = 50
-		melee_damage_upper = 50
-		armour_penetration = 100
-		obj_damage = 0
-		environment_smash = 0
-		new /obj/effect/overlay/temp/guardian/phase/out(get_turf(src))
-		alpha = 15
-		if(!forced)
-			src << "<span class='danger'><B>You enter stealth, empowering your next attack.</span></B>"
-		updatestealthalert()
-		toggle = TRUE
-	else if(!forced)
-		src << "<span class='danger'><B>You cannot yet enter stealth, wait another [max(round((stealthcooldown - world.time)*0.1, 0.1), 0)] seconds!</span></B>"
-
-/mob/living/simple_animal/hostile/guardian/assassin/proc/updatestealthalert()
-	if(stealthcooldown <= world.time)
-		if(toggle)
-			if(!instealthalert)
-				instealthalert = throw_alert("instealth", /obj/screen/alert/instealth)
-				clear_alert("canstealth")
-				canstealthalert = null
-		else
-			if(!canstealthalert)
-				canstealthalert = throw_alert("canstealth", /obj/screen/alert/canstealth)
-				clear_alert("instealth")
-				instealthalert = null
-	else
-		clear_alert("instealth")
-		instealthalert = null
-		clear_alert("canstealth")
-		canstealthalert = null
+	abilities = list(/datum/guardian_abilities/timestop)
```

Comment:
I disagree with defining these lists in the definitions. They should really be moved to Initialize